tiny_cnn/layers/layer.fwd.h

```diff
@@ -0,0 +1,336 @@
+/*
```

Comment:
@bhack @nyanp 
I had to adapt layer class into different files in order to avoid errors during forward declaration.

It's the same behaviour as in normal projects where you have class definition in *.hpp its implementation in *.cpp.
Here definition is in `layer.fwd.h` and implementation in `layer.h`